arch/arm/core/aarch64/userspace.S

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2020 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <toolchain.h>
+#include <linker/sections.h>
+#include <offsets_short.h>
+#include <arch/cpu.h>
+#include <syscall.h>
+#include "macro_priv.inc"
+
+_ASM_FILE_PROLOGUE
+
+/*
+ * size_t arch_user_string_nlen(const char *s, size_t maxsize, int *err_arg)
+ */
+
+GTEXT(z_arm64_user_string_nlen_fault_start)
+GTEXT(z_arm64_user_string_nlen_fault_end)
+GTEXT(z_arm64_user_string_nlen_fixup)
+
+GTEXT(arch_user_string_nlen)
+SECTION_FUNC(TEXT, arch_user_string_nlen)
+
+	mov	x3, x0
+	mov	x0, #0
+	mov	x4, #0
+
+strlen_loop:
+
+	cmp	x0, x1
+	beq	strlen_done
+
+z_arm64_user_string_nlen_fault_start:
+	ldrb	w5, [x3, x0]
```

Comment:
You probably want an ISB in here somewhere.

As-is, an SError may not be detected inside this region.